src/components/timeout-dialog/timeout-dialog.js

```diff
@@ -143,6 +137,9 @@ function TimeoutDialog($module, $sessionActivityService) {
 
     cleanupFunctions.push(() => {
       window.clearTimeout(timeout);
+      if (currentTimer) {
+        window.clearTimeout(currentTimer);
+      }
```

Comment:
moving the cleanup here, means that a cleanup function to clear the current timer is created each time the dialog is opened. currentTimer is created in the recursive function called a the end of setupDialog